sql/src/main/java/io/crate/protocols/postgres/types/PGTypes.java

```diff
@@ -97,31 +102,46 @@
         return TYPES;
     }
 
-    public static DataType fromOID(int oid) {
+    public static DataType<?> fromOID(int oid) {
         return PG_TYPES_TO_CRATE_TYPE.get(oid);
     }
 
-    public static PGType get(DataType type) {
-        if (type.id() == ArrayType.ID) {
-            DataType<?> innerType = ((ArrayType) type).innerType();
-            if (innerType.id() == ArrayType.ID) {
-                // if this is a nested collection stream it as JSON because
-                // postgres binary format doesn't support multidimensional arrays
-                // with sub-arrays of different length
-                // (something like [ [1, 2], [3] ] is not supported)
+    public static PGType get(DataType<?> type) {
+        switch (type.id()) {
```

Comment:
Do we need to handle the array of row type separately, such as we have a special handling for [RowType](https://github.com/crate/crate/pull/9627/files#diff-4af35b46ab2acffaf1a354f41e708866R133).

Reply:
Good point. Adding a fixup